sdk/Pulumi/Resources/ComponentResource.cs

```diff
@@ -52,6 +53,49 @@ public ComponentResource(string type, string name, ComponentResourceOptions? opt
             this.remote = remote;
         }
 
+        internal static Dictionary<string, object?> CollectOutputProperties(ComponentResource instance)
```

Comment:
I'd rather this stayed protected, or even better just folded this code into the RegisterOutputs() method. We should work out how to test this via a mock stack looking at resource outputs, that's what users will want to do.

Reply:
> Really? Can't you test this in a mock stack?

Stacks already do their own reflection logic that calls `RegisterOutputs(outputs)` internally which is why I went with this.

Reply:
Sure, but users will want to test their component outputs via the mock test system so lets make sure that works.

Reply:
I've extended `IMocks` now to support `RegisterResourceOutputs` which makes the code unit testable the way users would want to test their outputs. Folder the code back into `RegisterOutputs()` and refactored the unit test.

> `IMocks` has a default implementation of `RegisterResourceOutputs` that returns `Task.CompletedTask` so it doesn't break existing implementations